File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int cnt = 0;
        
        for(int[] command : commands){
            int i = command[0] - 1;
            int j = command[1];
            int k = command[2] - 1;
            
            int[] slide = Arrays.copyOfRange(array, i, j);
            Arrays.sort(slide);
            answer[cnt] = slide[k];
            cnt++;
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.go
import (
    "sort"
)

func solution(array []int, commands [][]int) []int {
    result := []int{}

    for _, cmd := range commands {
        i, j, k := cmd[0], cmd[1], cmd[2]

        sliced := make([]int, j-i+1)
        copy(sliced, array[i-1:j])

        sort.Ints(sliced)

        result = append(result, sliced[k-1])
    }

    return result
}
